src/components/ListItem.jsx

```diff
@@ -1,5 +1,43 @@
+import { useEffect, useState } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	itemId,
+	listPath,
+	purchaseDate,
+	nextPurchaseDate,
+}) {
+	let [isPurchased, setIsPurchased] = useState(false);
+
+	useEffect(() => {
+		const currentDate = new Date();
+		const nextPurchase = nextPurchaseDate.toDate();
+		if (purchaseDate) {
+			if (currentDate < nextPurchase) {
+				setIsPurchased(true);
+			} else {
+				setIsPurchased(false);
+			}
+		}
+	}, []);
+
+	const handleChange = async () => {
+		setIsPurchased(!isPurchased);
+		if (!isPurchased) {
+			await updateItem(listPath, itemId);
+		}
+	};
+	return (
+		<li className="ListItem">
+			<input
+				type="checkbox"
+				id={`checkbox-${itemId}`}
```

Comment:
Love seeing this! It’s a great approach to ensure each checkbox has a unique identifier, which is crucial for accessibility and linking with the corresponding <label>! 🚀